File: javascript/chapter9/dice.js
class Dice {
	/** @param {number} sides */
	constructor(sides) {
		this.sides = sides;
	}

	rollDie() {
		console.log(Math.floor(Math.random() * this.sides + 1));
	}
}

const dice = new Dice(6);
for (let i = 1; i <= 10; i++) {
	dice.rollDie();
}
const dice10 = new Dice(10);
for (let i = 1; i <= 10; i++) {
	dice10.rollDie();
}

const dice20 = new Dice(20);
for (let i = 1; i <= 10; i++) {
	dice20.rollDie();
}

File: python/chapter9/dice.py
import random


class Dice:
    def __init__(self, sides: int) -> None:
        self.sides = sides

    def roll_die(self):
        print(random.randint(1, self.sides) + 1)


dice = Dice(6)
for i in range(11):
    dice.roll_die()

dice10 = Dice(10)
for i in range(10):
    dice10.roll_die()

dice20 = Dice(20)
for i in range(10):
    dice20.roll_die()
